lib/mix/tasks/beacon.gen.site.ex

```diff
@@ -101,6 +104,7 @@ if Code.ensure_loaded?(Igniter) do
 
       port = Keyword.get_lazy(options, :port, fn -> Enum.random(4101..4999) end)
       secure_port = Keyword.get_lazy(options, :secure_port, fn -> Enum.random(8444..8999) end)
+      site_endpoint = Keyword.get_lazy(options, :endpoint, fn -> site_endpoint_module!(igniter, site) end)
```

Comment:
Values given from args are in binary so you need to convert to a proper module:

```suggestion
      site_endpoint = Keyword.get_lazy(options, :endpoint, fn -> site_endpoint_module!(igniter, site) end)
      site_endpoint = Module.concat([site_endpoint])
```